[TECH] Dans Pix API supprimer les notions erronées ou obsolètes de scope et de target utilisées pour la vérification du droit d’accès d’un utilisateur à une application (PIX-15945)

lego-technix opened the pull request:
## :pancakes: Problème

Les usecases `authenticateUser` et `authenticateOidcUser` se basent encore sur des notions de `scope` et de `target` qui sont erronées ou maintenant obsolètes et que nous voulons supprimer du code de Pix.

De plus, les usecases `authenticateUser` et `authenticateOidcUser` sont très proches mais ont du code légèrement différent, ce qui empêchent du travail de factorisation présent et futur.

## :bacon: Proposition

* Rendre le code des 2 usecases `authenticateUser` et `authenticateOidcUser` plus cohérent en utilisant les mêmes noms de variables et les mêmes noms de fonction,
* Utiliser une nouvelle notion générique de `RequestedApplication`, déduite à partir de l'_origin_, pour remplacer les notions de `scope` et de `target`.

PS : On aurait aimé utiliser un security prehandler en commun, mais ce n’est pas possible car dans le cas du usecase  `authenticateOidcUser` l'utilisateur n'est trouvé qu'à l'intérieur du usecase lors des échanges OIDC.


## 🧃 Remarques

RAS


## :yum: Pour tester

### Tester la connexion par login + mot de passe

Ces tests permettent de valider le bon fonctionnement du usecase `authenticateUser`.

1. Se connecter sur Pix App avec un utilisateur ayant un compte et constater l'authentification réussie,
2. Se connecter sur Pix Orga avec n'importe quel utilisateur n'ayant aucun droit sur des orgas (par exemple `hermione@school.net`) et constater l'échec de l'authentification avec l'affichage du message : 

   > L'accès à Pix Orga est limité aux membres invités. Chaque espace est géré par un administrateur Pix Orga propre à l'organisation qui l'utilise. Contactez-le pour qu'il vous y invite`

3. Se connecter sur Pix Orga avec un utilisateur ayant des droits sur une orga (par exemple `allorga@example.net`) et constater l'authentification réussie,
4. Se connecter sur Pix Admin avec n'importe quel utilisateur n'ayant aucun droit sur Pix Admin (par exemple `hermione@school.net`) et constater l'échec de l'authentification avec l'affichage du message : 

   > Vous n'avez pas les droits pour vous connecter.

5. Se connecter sur Pix Admin avec un utilisateur `SUPER_ADMIN` (par exemple `superadmin@example.net`) et constater l'authentification réussie.
 
### Tester la connexion par SSO

Ces tests permettent de valider le bon fonctionnement du usecase `authenticateOidcUser`.

1. Se connecter sur Pix App .fr par SSO avec un utilisateur approprié,
2. Se connecter sur Pix App .org par SSO avec un utilisateur approprié,
3. Se connecter sur Pix Admin par SSO avec un utilisateur approprié.

lego-technix committed:
refactor(api): make authenticateUser and authenticateOidcUser usecases more consistent

in variable and function naming to prepare for factoring out.

lego-technix committed:
test(api): remove controller success unit tests already tested as acceptance tests

lego-technix committed:
refactor(api): make authenticateOidcUser usecase use requestedApplication instead of target

lego-technix committed:
refactor(api): make authenticateUser usecase use requestedApplication instead of erroneous scope

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr11513.review.pix.fr)
  * [App (.org)](https://app-pr11513.review.pix.org)
  * [Orga (.fr)](https://orga-pr11513.review.pix.fr)
  * [Orga (.org)](https://orga-pr11513.review.pix.org)
  * [Certif (.fr)](https://certif-pr11513.review.pix.fr)
  * [Certif (.org)](https://certif-pr11513.review.pix.org)
  * [Junior](https://junior-pr11513.review.pix.fr)
  * [Admin](https://admin-pr11513.review.pix.fr)
  * [API](https://api-pr11513.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr11513.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr11513.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr11513/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr11513/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr11513/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr11513/environment)

lego-technix committed:
refactor(api): remove now unused constants

lego-technix committed:
fix(high-level-tests): issue HTTP requests to the fronts and never to the API directly

because otherwise it looses the front URL and prevent URL-based security
mechanisms to apply.

lego-technix committed:
test(api): factor out server calls in token.route acceptance tests

by using a _getPostFormOptions helper function.

P-Jeremy commented:
Test func 1 OK ✅

Test func 2 OK ✅

Test func 3 OK ✅

bpetetot reviewed (approved):
✅ Revu et testé avec succès

EmmanuelleBonnemay reviewed (approved):
Lu et testé avec succès

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr11513.review.pix.fr)
  * [App (.org)](https://app-pr11513.review.pix.org)
  * [Orga (.fr)](https://orga-pr11513.review.pix.fr)
  * [Orga (.org)](https://orga-pr11513.review.pix.org)
  * [Certif (.fr)](https://certif-pr11513.review.pix.fr)
  * [Certif (.org)](https://certif-pr11513.review.pix.org)
  * [Junior](https://junior-pr11513.review.pix.fr)
  * [Admin](https://admin-pr11513.review.pix.fr)
  * [API](https://api-pr11513.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr11513.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr11513.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr11513/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr11513/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr11513/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr11513/environment)

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr11513.review.pix.fr)
  * [App (.org)](https://app-pr11513.review.pix.org)
  * [Orga (.fr)](https://orga-pr11513.review.pix.fr)
  * [Orga (.org)](https://orga-pr11513.review.pix.org)
  * [Certif (.fr)](https://certif-pr11513.review.pix.fr)
  * [Certif (.org)](https://certif-pr11513.review.pix.org)
  * [Junior](https://junior-pr11513.review.pix.fr)
  * [Admin](https://admin-pr11513.review.pix.fr)
  * [API](https://api-pr11513.review.pix.fr/api/)
  * [API MaDDo](https://pix-api-maddo-review-pr11513.osc-fr1.scalingo.io/api/)
  * [Audit Logger](https://pix-audit-logger-review-pr11513.osc-fr1.scalingo.io/api/)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr11513/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr11513/environment)
  * [scalingo api-maddo](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-maddo-review-pr11513/environment)
  * [scalingo audit-logger](https://dashboard.scalingo.com/apps/osc-fr1/pix-audit-logger-review-pr11513/environment)
